File: api/test2.php
<?php

//
// $Id$
//

require ( "sphinxapi.php" );

$docs = array
(
	"this is my test text to be highlighted, and for the sake of the testing we need to pump its length somewhat",
	"another test text to be highlighted, below limit",
	"test number three, without phrase match",
	"final test, not only without phrase match, but also above limit and with swapped phrase text test as well",
);
$words = "test text";
$index = "test1";
$opts = array
(
	"before_match"		=> "<b>",
	"after_match"		=> "</b>",
	"chunk_separator"	=> " ... ",
	"limit"				=> 60,
	"around"			=> 3,
);

foreach ( array(0,1) as $exact )
{
	$opts["exact_phrase"] = $exact;
	print "exact_phrase=$exact\n";

	$cl = new SphinxClient ();
	$res = $cl->BuildExcerpts ( $docs, $index, $words, $opts );
	if ( !$res )
	{
		die ( "ERROR: " . $cl->GetLastError() . ".\n" );
	} else
	{
		$n = 0;
		foreach ( $res as $entry )
		{
			$n++;
			print "n=$n, res=$entry\n";
		}
		print "\n";
	}
}

//
// $Id$
//

?>

File: api/test2.py
#
# $Id$
#
from __future__ import print_function
from sphinxapi import *
import sys

docs = ['this is my test text to be highlighted','this is another test text to be highlighted']
words = 'test text'
index = 'forum'

opts = {'before_match':'<b>', 'after_match':'</b>', 'chunk_separator':' ... ', 'limit':400, 'around':15}

cl = SphinxClient()
res = cl.BuildExcerpts(docs, index, words, opts)

if not res:
	print  ('ERROR: %s\n' % cl.GetLastError())
else:
	n = 0
	for entry in res:
		n += 1
		print ('n=%d, res=%s' % (n, entry))

#
# $Id$
#
